### PR TITLE
Move LabelSet implementation from API to SDK

### PR DESCRIPTION
**Pull request by cijothomas**
LabelSet implementation is done inside API which is incorrect. This PR moves the implementation to the SDK.

### CODE REVIEW

File: src/OpenTelemetry.Api/Metrics/BlankLabelSet.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace OpenTelemetry.Metrics
+{
+    /// <summary>
+    /// Blank or no-op labelset.
+    /// </summary>
+    public sealed class BlankLabelSet : LabelSet
+    {
+        /// <inheritdoc/>
+        public override IEnumerable<KeyValuePair<string, string>> Labels { get; set; }
```
Comment (bruno-garcia) [inline src/OpenTelemetry.Api/Metrics/BlankLabelSet.cs]: This needs to override? The abstract class only has a getter and setter too

File: src/OpenTelemetry.Api/Metrics/LabelSet.cs [hunk 2]
```diff
@@ -21,73 +21,17 @@ namespace OpenTelemetry.Metrics
...
-
-                encoder.Append(label.Key);
-                encoder.Append("\t");
-                encoder.Append(label.Value);
-                isFirstLabel = false;
-            }
-
-            return encoder.ToString();
-        }
+        public abstract IEnumerable<KeyValuePair<string, string>> Labels { get; set; }
```
Comment (bruno-garcia) [inline src/OpenTelemetry.Api/Metrics/LabelSet.cs]: Would it make sense to leave an instance here?

```suggestion
        public abstract IEnumerable<KeyValuePair<string, string>> Labels { get; set; } = Enumerable.Empty<KeyValuePair<string, string>>();
```

File: src/OpenTelemetry.Api/Metrics/BlankLabelSet.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Metrics/BlankLabelSet.cs]: @bruno-garcia Took your suggestion and made abstract class provide Empty implementation for the same.

File: src/OpenTelemetry/Metrics/LabelSetSDK.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace OpenTelemetry.Metrics
+{
+    /// <summary>
+    /// LabelSet implemetation.
+    /// </summary>
+    public class LabelSetSDK : LabelSet
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry/Metrics/LabelSetSDK.cs]: is it needs to be public?

File: src/OpenTelemetry.Api/Metrics/BlankLabelSet.cs (hunk 1 above)
Comment (bruno-garcia) [reply src/OpenTelemetry.Api/Metrics/BlankLabelSet.cs]: Thanks @cijothomas !

**Comment by cijothomas** [comment]
@SergeyKanzhelev please merge

**Pull request merged**